File: 3sem/poo/Aula01-Codigo/Java/Exemplo5-HerancaPolimorfismo/src/exemplo4herancapolimorfismo/Quadrado.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package exemplo4herancapolimorfismo;

/**
 *
 * @author Junio
 */
public class Quadrado extends Forma {

    Quadrado(String umNome, float umParam){
        super(umNome, umParam);
    }

    public float CalculaArea(){
        return (float) (fParam * fParam);
    }
}

File: 3sem/poo/Aula01-Codigo/C++/Exemplo05-FormaGeometrica/Quadrado.cpp
#include <iostream>
#include <string>
#include "FormaGeometrica.cpp"
using namespace std;

class Quadrado : public FormaGeometrica {
  public:
    Quadrado(string umNome, float umParam) : FormaGeometrica(umNome, umParam){

    };

    void MostraNome(){
        cout << sNome << " diz: - Quadrado mesmo." << "\n";
    };

    float CalculaArea(){
        return (float) (fParam * fParam);
    };
};
